src/edu/csus/ecs/pc2/core/LanguageUtilities.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.core;
+
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.model.Language;
+
+/**
+ * Utility methods for languages
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ */
+public class LanguageUtilities {
+    
+    /**
+     * List of language extensions and their full or partial language display name
+     */
+    public static final  String[][] LANGUAGE_LIST = { //
+            { "cpp", "C++" }, //
+            { "cc", "C++" }, //
+            { "C", "C++" }, //
+            { "c++", "C++" }, //
+            { "py", "Python" }, //
+            { "cs", "Mono" }, //
+            { "pl", "Perl" }, //
+
+    };
+
+
+    /**
+     * get file extension.
+     * 
+     * @param filename
+     * @return file extension if no period found returns null
+     */
+    public static String getExtension(String filename) {
+        int idx = filename.indexOf(".");
+        if (idx != -1) {
+            return filename.substring(idx + 1, filename.length());
+        }
+        return null;
+    }
```

Comment:
It appears to me that the above function will fail (that is, will not return the proper file extension) if the specified `filename` includes more than one "." (e.g. a file named `my.shortprog.c`).  
I think the method should have used `filename.lastIndexOf(".")` rather than `filename.indexOf(".")`